src/Tools/dotnet-monitor/Egress/Configuration/OptionsTypeToProviderTypesMapper.cs

```diff
@@ -53,14 +44,13 @@ public IEnumerable<IConfigurationSection> GetProviderSections(Type optionsType)
         }
 
         /// <summary>
-        /// Determines if the given section is a built in type property/field of <see cref="EgressOptions"/>.
+        /// Determines if the given section is a built in type property/field of the Egress configuration section.
         /// </summary>
         /// <param name="s">The <see cref="IConfigurationSection"/> to evaluate, this should be one of the children of the "Egress" section.</param>
         /// <returns><see langword="true"/> if and only if the given section is one of of the pre-defined</returns>
         private static bool IsBuiltInSection(IConfigurationSection s)
         {
-            Type egressType = typeof(EgressOptions);
-            return egressType.GetProperty(s.Key) != null;
+            return ConfigurationKeys.Egress_Properties == s.Key;
```

Comment:
If I'm reading this correctly, we're just checking if the key is "Properties" since all of the providers are now out of the box - if that's the case, perhaps either the commenting/naming of this method could be updated to make that more transparent.